graphix/pattern.py

```diff
@@ -1527,14 +1546,65 @@ def to_qasm3(self, filename):
 
     def copy(self) -> Pattern:
         result = self.__new__(self.__class__)
-        result.__seq = [cmd.model_copy() for cmd in self.__seq]
+        result.__seq = [cmd.model_copy(deep=True) for cmd in self.__seq]
         result.__input_nodes = self.__input_nodes.copy()
         result.__output_nodes = self.__output_nodes.copy()
         result.__n_node = self.__n_node
         result._pauli_preprocessed = self._pauli_preprocessed
         result.results = self.results.copy()
         return result
 
+    def __extract_pauli_nodes(self, leave_nodes: set[int] | None = None) -> dict[int, tuple[command.M]]:
```

Comment:
`tuple[command.M]` is almost identical to `command.M`. Why not use `command.M`?

Reply:
Fixed in 8e92b97: it was a mistake indeed, the return type is now `dict[int, command.M]`.